analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxFacade.cs

```diff
@@ -53,6 +54,18 @@ internal sealed class CSharpSyntaxFacade : SyntaxFacade<SyntaxKind>
 
         public override bool IsNullLiteral(SyntaxNode node) => node.IsNullLiteral();
 
+        public override IEnumerable<SyntaxNode> ArgumentExpressions(SyntaxNode node) =>
+            node switch
+            {
+                ObjectCreationExpressionSyntax objectCreation => ArgumentExpressions(objectCreation),
+                null => Enumerable.Empty<SyntaxNode>(),
+                var _ when ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(node) => Enumerable.Empty<SyntaxNode>(),
+                _ => throw InvalidOperation(node, nameof(ArgumentExpressions))
+            };
+
+        private static IEnumerable<SyntaxNode> ArgumentExpressions(ObjectCreationExpressionSyntax node) =>
+            node.ArgumentList?.Arguments.Select(x => x.Expression) ?? Enumerable.Empty<SyntaxNode>();
```

Comment:
Let's embed it. It's more clear that this thing is not meant to return null.
```suggestion
                ObjectCreationExpressionSyntax objectCreation => objectCreation.ArgumentList?.Arguments.Select(x => x.Expression) ?? Enumerable.Empty<SyntaxNode>(),
                null => Enumerable.Empty<SyntaxNode>(),
                var _ when ImplicitObjectCreationExpressionSyntaxWrapper.IsInstance(node) => Enumerable.Empty<SyntaxNode>(),
                _ => throw InvalidOperation(node, nameof(ArgumentExpressions))
            };
```